src/index.js

```diff
@@ -1,45 +1,200 @@
+        const resultJSON = JSON.parse(result);
+        if (!resultJSON.error) self.setUserAccessToken(resultJSON);
+        return resultJSON;
+    });
+};
+
+/**
+ * Use a refresh token to update a User access token (Updating the expired access token)
+ *
+ * @param refreshToken refresh token, defaults to pre-assigned refresh token
+ * @param scopes array of scopes for the access token
+ * @return userAccessToken object (without refresh_token)
+*/
+const getUserTokenByRefresh = function (refreshToken = null) {
+    if (!this.options.clientID) throw new Error('Missing Client ID');
+    if (!this.options.clientSecret) throw new Error('Missing Client Secret or Cert Id');
+    if (!this.options.body) throw new Error('Missing Body, required Grant type');
+    if (!refreshToken && !this.options.refreshToken) {
+        throw new Error('Refresh token is required, to generate refresh token use getUserTokenByCode method'); // eslint-disable-line max-len
+    }
+    refreshToken = refreshToken ? refreshToken : this.options.refreshToken;
+    let scopesParam = this.options.body.scopes
+        ? Array.isArray(this.options.body.scopes)
+            ? this.options.body.scopes.join('%20')
+            : this.options.body.scopes
+        : DEFAULT_API_SCOPE;
+    this.options.data = qs.stringify({
+        refresh_token: refreshToken,
+        grant_type: 'refresh_token',
+        scope: scopesParam
+    });
+    this.options.contentType = 'application/x-www-form-urlencoded';
+    const self = this;
+    const encodedStr = commonUtils.base64Encode(`${this.options.clientID}:${this.options.clientSecret}`);
+    const auth = `Basic ${encodedStr}`;
+    return makeRequest(this.options, '/identity/v1/oauth2/token', 'POST', auth).then(result => {
+        const resultJSON = JSON.parse(result);
+        if (!resultJSON.error) self.setUserAccessToken(resultJSON);
+        return resultJSON;
+    });
+};
+
+/**
+ * Assign user access token and refresh token returned from authorization grant workflow (i.e getUserTokenByRefresh)
+ *
+ * @param userAccessToken userAccessToken obj returned from getUserTokenByCode or getAccessTokenByRefresh
+*/
+const setUserAccessToken = function (userAccessToken) {
+    if (!userAccessToken.token_type === 'User Access Token') throw new Error('userAccessToken is either missing or invalid');
+    if (userAccessToken.refresh_token) this.options.refreshToken = userAccessToken.refresh_token;
+    this.options.userAccessToken = userAccessToken.access_token;
+};
+
+/**
+ * Assign application access token returned from client credentials workflow (i.e getAccessToken)
+ *
+ * @param appAccessToken appAccessToken obj returned from getApplicationToken
+*/
+const setAppAccessToken = function (appAccessToken) {
+    if (!appAccessToken.token_type === 'Application Access Token') throw new Error('appAccessToken is either missing or invalid');
+    this.options.appAccessToken = appAccessToken.access_token;
+};
+
```

Comment:
@TotallyNotChase  Shall we keep it in a new file as these methods are related to access tokens like file `credentials.js`?.